web/cat/mad_hatter/decorators.py

```diff
@@ -7,11 +7,20 @@
 
 
 log(tool)
+hooks = []
```

Comment:
This does not look good to me. Hooks array should live inside the MadHatter and Hooks do not need to know that they are part of an array